docs/topics/upgrade.md

```diff
@@ -0,0 +1,77 @@
+
+This guide assumes you already have deployed a cluster using `aks-engine`. For more details on how to do that see [deploy](../tutorials/deploy.md).
+
+## Upgrade
+
+This document provides guidance on how to upgrade the Kubernetes version for an existing `aks-engine` cluster and recommendations for adopting `aks-engine upgrade` as a production quality tool.
+
+<a name="pre-requirements"></a>
+
+### Know before you go
+
+In order to ensure that your `aks-engine upgrade` operation runs smoothly, there are a few things you should be aware of before getting started.
+
+1) You will need access to the original `_output/` directory files that were generated during cluster deployment in order to run `upgrade`.  `aks-engine` will use the `--deployment-dir` argument provided in the upgrade command to introspect the Azure Resource Manager (ARM) template and parameters to determine the cluster's current Kubernetes version and its current state.
+
+2) `aks-engine upgrade` expects a cluster configuration that conforms to the current state of the cluster. In other words, the Azure resources deployed by `aks-engine` should be in the same state that when they were originally created by `aks-engine`. If you perform manual operations on your Azure IaaS resources (other than `aks-engine scale` and `aks-engine upgrade`) you risk not being able to upgrade your cluster successfully as `aks-engine` won't be able to reconcile the cluster's current state. This includes naming of resources; `aks-engine` relies on some resources (such as VMs) to be named in accordance with the original `aks-engine` deployment.
+
+3) `aks-engine upgrade` allows one subsequent minor version upgrade at a time, as long as the next minor version is supported. If the next minor version is deprecated, `aks-engine` allows upgrade to the next supported minor version.
+
+To get the list of all available Kubernetes versions and upgrades, run the `orchestrators` command and specify Kubernetes orchestrator type:
+
+```bash
+./bin/aks-engine orchestrators --orchestrator Kubernetes
+```
+
+To get the information specific to the cluster, provide its current Kubernetes version in the `version` arg:
+```bash
+./bin/aks-engine orchestrators --orchestrator Kubernetes --version 1.11.5
+```
+
+4) If using `aks-engine upgrade` in production, it is recommended to run a dry run test on an identical cluster before performing the upgrade, especially if the cluster configuration is "unusual", or in other words differs significantly from defaults. The reason for this is that `aks-engine` supports many different cluster configurations and the extent of E2E testing that the AKS-Engine team runs cannot possibly cover every single configuration out there. Therefore, it is recommended that you make sure that your specific cluster configuration works with the existing upgrade implementation before starting this long-running operation.
```

Comment:
s/it is recommended to run a dry run test on an identical cluster/it is recommended to stage an upgrade test on an cluster that was built to the same specifications (built with the same cluster configuration + aks-engine version) as your production cluster/

Reply:
s/"unusual"/"interesting"/